server/lib/accounts.js

```diff
@@ -63,8 +63,13 @@ Accounts.onCreateUser(function(options, user = {}) {
 	user.active = !RocketChat.settings.get('Accounts_ManuallyApproveNewUsers');
 
 	if (!user.name) {
-		if (options.profile && options.profile.name) {
-			user.name = options.profile.name;
+		if (options.profile) {
+			if (options.profile.name) {
+				user.name = options.profile.name;
+			} else if (options.profile.firstName && options.profile.lastName) {
```

Comment:
can it happen to have the `firstName` but not the `lastName`?

If so, it would be better to split the `if` in two so the user will got at least the `firstName`